spyder_notebook/widgets/client.py

```diff
@@ -204,6 +204,26 @@ def save(self):
         """Save current notebook."""
         self.notebookwidget.click("#save-notbook button")
 
+    def get_session_url(self):
+        """Get the kernel sessions url of the client."""
+        return self.add_token(url_path_join(self.server_url, 'api/sessions'))
+
+    def get_kernel_id(self):
+        """Get the kernel id of the client."""
+        sessions_url = self.get_session_url()
+        sessions_req = requests.get(sessions_url).content.decode()
+        sessions = json.loads(sessions_req)
+
+        if os.name == 'nt':
+            path = self.path.replace('\\', '/')
+        else:
+            path = self.path
+
+        for session in sessions:
+            if session['notebook']['path'] == path:
+                kernel_id = session['kernel']['id']
+                return kernel_id
+
     def shutdown_kernel(self):
         """Shutdown the kernel of the client."""
         sessions_url = self.add_token(url_path_join(self.server_url,
```

Comment:
Sorry, we can simplify this whole function by simply calling `self.get_kernel_id()`.

Please do it so we can remove this chunk of repeated code here.